cirq/study/trial_result.py

```diff
@@ -273,12 +273,14 @@ def __eq__(self, other):
         return self.data.equals(other.data) and self.params == other.params
 
     def _measurement_shape(self):
-        return self.params, {k: v.shape[1] for k, v in self.measurement.items()}
+        return self.params, {
+            k: v.shape[1] for k, v in self.measurements.items()
+        }
 
     def __add__(self, other: 'cirq.TrialResult') -> 'cirq.TrialResult':
         if not isinstance(other, type(self)):
             return NotImplemented
-        if self._measurement_shape() != other._measurement_shape():
+        if self._measurement_shape() == other._measurement_shape():
```

Comment:
Flip this into a guard statement. "If something is bad then fail", not "if something is good then do the whole thing" followed by a raise at the end.